source/renderer/app/components/wallet/receive/Address.js

```diff
@@ -5,43 +5,132 @@ import CopyToClipboard from 'react-copy-to-clipboard';
+    window.addEventListener('resize', this.updateWindowDimensions);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateWindowDimensions);
+  }
+
+  updateWindowDimensions = () => {
+    const { currentBreakPoint } = this.state;
+    const newBreakpoint = this.getBreakpoint(window.innerWidth);
+    if (currentBreakPoint !== newBreakpoint) {
+      const { minCharsInit, minCharsEnd } = BREAKPOINTS[newBreakpoint];
+      this.setState({
+        currentBreakPoint: newBreakpoint,
+        minCharsInit,
+        minCharsEnd,
+      });
+    }
+  };
+
+  getBreakpoint = (windowWidth: number) => {
+    if (windowWidth >= 1081) return 4;
+    if (windowWidth >= 1050) return 3;
+    if (windowWidth >= 1000) return 2;
+    if (windowWidth >= 950) return 1;
+    return 0;
+  };
+
   render() {
-    const { address, onCopyAddress, copyAddressLabel, index } = this.props;
+    const {
+      address,
+      onShareAddress,
+      onCopyAddress,
+      shareAddressLabel,
+      copyAddressLabel,
+      isIncentivizedTestnet,
+    } = this.props;
     const addressClasses = classnames([
-      `generatedAddress-${index + 1}`,
+      `receiveAddress-${address.id}`,
       styles.component,
       address.used ? styles.usedWalletAddress : null,
     ]);
+    const { minCharsInit, minCharsEnd } = this.state;
     return (
       <div className={addressClasses}>
         <div className={styles.addressId} id={`address-${address.id}`}>
-          {address.id}
+          {ellipsis(address.id, minCharsInit, minCharsEnd)}
         </div>
         <div className={styles.addressActions}>
-          <CopyToClipboard
-            text={address.id}
-            // eslint-disable-next-line react/jsx-no-bind
-            onCopy={onCopyAddress.bind(this, address.id)}
-          >
-            <span className={styles.copyAddress}>
-              <SVGInline svg={iconCopy} className={styles.copyIcon} />
-              <span className={styles.copyAddressLabel}>
-                {copyAddressLabel}
+          {!isIncentivizedTestnet ? (
+            <button
+              className={styles.shareAddressButton}
+              onClick={() => onShareAddress(address)}
+            >
+              <SVGInline svg={iconQR} className={styles.shareIcon} />
+              <span className={styles.shareAddressLabel}>
+                {shareAddressLabel}
+              </span>
+            </button>
+          ) : null}
+          {isIncentivizedTestnet ? (
+            <CopyToClipboard
+              text={address.id}
+              onCopy={() => onCopyAddress(address.id)}
+            >
+              <span className={styles.copyAddress}>
+                <SVGInline svg={iconCopy} className={styles.copyIcon} />
+                <span className={styles.copyAddressLabel}>
+                  {copyAddressLabel}
+                </span>
               </span>
-            </span>
-          </CopyToClipboard>
+            </CopyToClipboard>
+          ) : null}
```

Comment:
@daniloprates you don't need 2 clauses with `null` fallback. You can put all together like:
`{isIncentivizedTestnet ? (...code with copy) : (... code with share)}`